docs/source/guides/speed.rst

```diff
@@ -0,0 +1,279 @@
+.. testsetup:: *
+
+    from pytorch_lightning.trainer.trainer import Trainer
+    from pytorch_lightning.callbacks.early_stopping import EarlyStopping
+    from pytorch_lightning.core.lightning import LightningModule
+
+.. _speed:
+
+#######################
+Speed up model training
+#######################
+
+There are multiple ways you can speed up your model.
+
+.. _early_stopping:
+
+**************
+Early stopping
+**************
+
+.. raw:: html
+
+    <video width="50%" max-width="400px" controls
+    poster="https://pl-bolts-doc-images.s3.us-east-2.amazonaws.com/pl_docs/trainer_flags/yt_thumbs/thumb_earlystop.png"
+    src="https://pl-bolts-doc-images.s3.us-east-2.amazonaws.com/pl_docs/yt/Trainer+flags+19-+early+stopping_1.mp4"></video>
+
+|
+
+Early stopping based on metric
+==============================
+The
+:class:`~pytorch_lightning.callbacks.early_stopping.EarlyStopping`
+callback can be used to monitor a validation metric and stop the training when no improvement is observed.
+
+To enable it:
+
+- Import :class:`~pytorch_lightning.callbacks.early_stopping.EarlyStopping` callback.
+- Log the metric you want to monitor using :func:`~pytorch_lightning.core.lightning.LightningModule.log` method.
+- Init the callback, and set `monitor` to the logged metric of your choice.
+- Pass the :class:`~pytorch_lightning.callbacks.early_stopping.EarlyStopping` callback to the :class:`~pytorch_lightning.trainer.trainer.Trainer` callbacks flag.
+
+.. code-block:: python
```

Comment:
This should be as minimal as possible it should explain what early stopping is how it speeds up training convergence and link to the relevant API Reference calls. The goal of this is not to be reference itself

Reply:
I don't think all this info is in the API reference though, where should it be then?